src/crypto/crypto_util.h

```diff
@@ -194,28 +199,15 @@ template <typename... Args>
 void CryptoErrorStore::Insert(const NodeCryptoError error, Args&&... args) {
   const char* error_string = nullptr;
   switch (error) {
-    case NodeCryptoError::CIPHER_JOB_FAILED:
-      error_string = "Cipher job failed";
-      break;
-    case NodeCryptoError::DERIVING_BITS_FAILED:
-      error_string = "Deriving bits failed";
-      break;
-    case NodeCryptoError::ENGINE_NOT_FOUND:
-      error_string = "Engine \"%s\" was not found";
-      break;
-    case NodeCryptoError::INVALID_KEY_TYPE:
-      error_string = "Invalid key type";
-      break;
-    case NodeCryptoError::KEY_GENERATION_JOB_FAILED:
-      error_string = "Key generation failed";
-      break;
-    case NodeCryptoError::OK:
-      error_string = "Ok";
-      break;
+#define V(CODE, DESCRIPTION) \
+    case NodeCryptoError::CODE: error_string = DESCRIPTION; break;
+    NODE_CRYPTO_ERROR_CODES_MAP(V)
+#undef V
   }
   errors_.emplace_back(SPrintF(error_string,
                                std::forward<Args>(args)...));
 }
+#undef NODE_CRYPTO_ERROR_CODES_MAP
```

Comment:
Having this available outside of the header might be useful at some point :woman_shrugging:

Reply:
@addaleax Agreed, exposed it in 6ce921aaef6aa35ab39d72f882663ec47059e573. Could you think of any place where it could be used currently? I could incorporate it in a follow-up PR.

Reply:
No, not right now, although I guess if we add more error codes to `CryptoErrorStore::Insert()` then it might make sense to move the part that generates `error_string`/that doesn’t use `args` back into a `.cc` file (because right now every call to `CryptoErrorStore::Insert()` potentially creates a new copy of the function).